### PR TITLE
requirements manager changed to use pipenv libraries

### PR DESCRIPTION
**Pull request by KPostOffice**
these changes are both to allow for all vcs sources as well as ensuring future compatibility as the capabilities of pipenv change

Signed-off-by: Kevin <kpostlet@redhat.com>

## Related Issues and Dependencies

fixes: #1142 

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by KPostOffice** [comment]
@harshad16 PTAL if you can

**Review by harshad16 — COMMENTED** [review]
lgtm
just one doubt

File: kebechet/managers/pipfile_requirements/pipfile_requirements.py [hunk 1]
```diff
@@ -135,22 +104,47 @@ def run(self, lockfile: bool = False) -> None:  # type: ignore
...
                 commit_msg="Update requirements.txt respecting requirements in {}".format(
                     "Pipfile" if not lockfile else "Pipfile.lock"
                 ),
                 branch_name=branch_name,
                 files=["requirements.txt"],
+                force_push=True,
             )
+            if not self.get_prs_by_branch(branch_name):
+                self.create_pr(
+                    title=f"Syncing requirements.txt with {'Pipfile.lock' if lockfile else 'Pipfile'}",
```
[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/managers/pipfile_requirements/pipfile_requirements.py:146]: "using"? maybe

File: kebechet/managers/pipfile_requirements/pipfile_requirements.py [hunk 2]
```diff
@@ -135,22 +104,47 @@ def run(self, lockfile: bool = False) -> None:  # type: ignore
...
+        try:
+            file_contents = self.project.get_file_content(
+                path="requirements.txt", ref=branch_name
+            )
+            requirements_txt_content = sorted(file_contents.splitlines())
+        except (FileNotFoundError, UnknownObjectException):
+            requirements_txt_content = []  # either branch or file does not exist.
+
+        if requirements == requirements_txt_content:
+            _LOGGER.info("requirements in PR are up to date")
```
[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/managers/pipfile_requirements/pipfile_requirements.py:127]: The conditionals are a bit different. One is checking whether the files need changed on master and the other is checking whether the current changes are up to date. I'll pull this into a separate function. Something with a signature like:

```
function does_remote_requirements_eql_new(self, requirements: list, ref: str = None)
    ref = self.project.default_branch if ref is None else ref
    ...
```

File: kebechet/managers/pipfile_requirements/pipfile_requirements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (harshad16) [reply kebechet/managers/pipfile_requirements/pipfile_requirements.py:146]: lets stay with , "with" :)

File: kebechet/managers/pipfile_requirements/pipfile_requirements.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (harshad16) [reply kebechet/managers/pipfile_requirements/pipfile_requirements.py]: ack, can we have some comment on top of the second time looping based on branch.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

Thanks :+1:

**Pull request merged**